### ISSUE TITLE
status

### ISSUE THREAD

**Issue by jamesdixon** [nelsonic/hapi-upload#2]
Hi, did anything ever come of this project? This was the type of thing I was looking for :)

**Comment by nelsonic** [comment]
@jamesdixon funny enough, we built it for a client (in [meteor](https://github.com/dwyl/image-uploads) ...) but then _forgot_ to translate it to here... because we have since decide to write it for AWS Lambda...

> what are your _exact_ requirements?

**Comment by jamesdixon** [comment]
@nelsonic, forgetting happens :) 

I have an Ionic mobile application and am using Hapi for my API. My requirements are fairly similar, but not exact:
- The API will accept one or images from the mobile application
- The API will optimize and auto-rotate the image(s) (if necessary)
- The API will upload the image(s) to Azure Storage and return their locations for storage in the database

**Comment by nelsonic** [comment]
Ok. cool. first two are solved. only the third one we will need to work on. 
are you tied to Azure by client/requirement?

**Comment by jamesdixon** [comment]
Yes, tied to Azure due to it being free for me for the next three years :) It appears they have a fairly decent Node SDK specifically for their Azure Storage product.

I haven't done any image uploading work with Hapi, so if you have any recommendations on the best approach, that would be great.

Thanks for your help!

**Comment by nelsonic** [comment]
@jamesdixon **_three years**_! how did you get _that_ deal...?

**Comment by jamesdixon** [comment]
It's their [Bizspark](https://www.microsoft.com/bizspark) program. $750/mo in credit for three years. It's enough to get my business off the ground with a pretty decent setup :+1:

**Comment by nelsonic** [comment]
Oh... wow.... that's pretty _aggressive_ customer acquisition!! fair enough...
Might have to investigate that... :+1:
